src/xb-value-bindings.h

```diff
@@ -0,0 +1,86 @@
+/* -*- Mode: C; tab-width: 8; indent-tabs-mode: t; c-basic-offset: 8 -*-
+ * vi:set noexpandtab tabstop=8 shiftwidth=8:
+ *
+ * Copyright (C) 2020 Endless OS Foundation LLC
+ *
+ * Author: Philip Withnall <withnall@endlessm.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#pragma once
+
+#include <glib.h>
+
+#include "xb-opcode.h"
+
+G_BEGIN_DECLS
+
+/**
+ * XbValueBindings:
+ *
+ * An opaque struct which contains values bound to a query.
+ *
+ * Since: 0.3.0
+ */
+typedef struct {
+	gint dummy0;
+	gpointer dummy1[2];
+	gint dummy2;
+	gpointer dummy3[2];
+	gint dummy4;
+	gpointer dummy5[2];
+	gint dummy6;
+	gpointer dummy7[2];
+	gpointer dummy8[3];
```

Comment:
this looks pretty l33t -- too l33t for me :)

Reply:
There’s no real alternative that I know of. The idea is that the structs in `xb-value-bindings.c` and `.h` mirror each other in terms of size and alignment, but the struct members are obscured (‘dummy’) in the public version, so that people can’t access the struct members directly. Because the struct is public, though, they can still stack-allocate it for speed. The internal C code casts from the public (‘dummy’) struct to the private (real) struct before interacting with the real struct members.

It’s not an uncommon pattern, and is used in various places in GLib, such as `GHashTableIter`.

Reply:
Don't we need a /*< private >*/ sprinkled in tho?

Reply:
Yes, good suggestion. And for `XbQueryContext`, which uses the same approach. I’ll do that later and then rebase everything ready to (maybe?) be merged. Do you have any other review comments?

Reply:
> Do you have any other review comments?

Nope! LGTM after that, thanks.

Reply:
OK, done for both, and I’ve squashed all the fixups in.

Reply:
MERGE! :) Thanks.